Windows CI Matrix only tests Python 3.12

Despite how it looks in GitHub, the Winows GitHub Actions only test the nuget package that are in the Project file, which is hard-coded to 3.12.4.

There's a complication - Python team uses different naming convention for their binaries and NuGet packages...

![image](https://github.com/user-attachments/assets/624be4e9-c1ce-49b4-8e39-daa2df155900)

Subsequently, an attempt to use the Python version set on CI results in MSBuild warnings:
```powershell
PS \src> $env:PYTHON_VERSION="3.13.0-beta.4"
PS \src> dotnet restore /bl
Restore succeeded with 3 warning(s) in 0.7s
    \src\Integration.Tests\Integration.Tests.csproj : warning NU1603: Integration.Tests depends on python (>= 3.13.0-beta.4) but python 3.13.0-beta.4 was not found. python 3.13.0-rc1 was resolved instead.
    \src\CSnakes.Runtime.Tests\CSnakes.Runtime.Tests.csproj : warning NU1603: CSnakes.Runtime.Tests depends on python (>= 3.13.0-beta.4) but python 3.13.0-beta.4 was not found. python 3.13.0-rc1 was resolved instead.
    \src\Profile\Profile.csproj : warning NU1603: Profile depends on python (>= 3.13.0-beta.4) but python 3.13.0-beta.4 was not found. python 3.13.0-rc1 was resolved instead.
```

--- a/.github/workflows/dotnet-ci.yml
+++ b/.github/workflows/dotnet-ci.yml
@@ -36,6 +36,7 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: "${{ matrix.python }}"
+
       - name: Setup .NET
         uses: actions/setup-dotnet@v4
         with:
@@ -52,9 +53,13 @@ jobs:
       - name: Restore dependencies
         run: dotnet restore
         working-directory: src
+        env:
+          PYTHON_VERSION: ${{ steps.installpython.outputs.python-version }}
+
       - name: Build
         run: dotnet build --no-restore
         working-directory: src
+
       - name: Test
         run: dotnet test --no-build --verbosity normal --collect "XPlat Code Coverage" --results-directory test-results --logger "trx;LogFileName=test-results-${{ matrix.os }}-${{ matrix.python }}-${{ matrix.dotnet }}.trx"
         working-directory: src

--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -1,0 +1,7 @@
+<Project>
+  <PropertyGroup>
+    <PythonVersion>3.12.4</PythonVersion>
+    <!-- See https://github.com/tonybaloney/CSnakes/issues/154#issuecomment-2352116849 -->
+    <PythonVersion Condition=" '$(PYTHON_VERSION)' != '' ">$(PYTHON_VERSION.Replace('alpha.','a').Replace('beta.','b').Replace('rc.','rc'))</PythonVersion>
+  </PropertyGroup>
+</Project>

--- a/samples/Aspire/CSnakesAspire.ApiService/CSnakesAspire.ApiService.csproj
+++ b/samples/Aspire/CSnakesAspire.ApiService/CSnakesAspire.ApiService.csproj
@@ -25,7 +25,7 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageReference Include="python" Version="3.12.4" Condition="'$(IsWindows)' == 'true'" />
+        <PackageReference Include="python" Version="$(PythonVersion)" Condition="'$(IsWindows)' == 'true'" />
     </ItemGroup>
 
     <ItemGroup>

--- a/samples/simple/QuickConsoleTest/QuickConsoleTest.csproj
+++ b/samples/simple/QuickConsoleTest/QuickConsoleTest.csproj
@@ -8,7 +8,7 @@
 
     <ItemGroup>
         <PackageReference Include="CSnakes.Runtime" Version="1.*-*" />
-        <PackageReference Include="python" Version="3.12.4" />
+        <PackageReference Include="python" Version="$(PythonVersion)" />
     </ItemGroup>
 
     <ItemGroup>

--- a/samples/simple/WebApp/WebApp.csproj
+++ b/samples/simple/WebApp/WebApp.csproj
@@ -12,7 +12,7 @@
 
     <ItemGroup>
         <PackageReference Include="CSnakes.Runtime" Version="1.*-*" />
-        <PackageReference Include="python" Version="3.12.4" />
+        <PackageReference Include="python" Version="$(PythonVersion)" />
     </ItemGroup>
 
 </Project>

--- a/src/CSnakes.Runtime.Tests/CSnakes.Runtime.Tests.csproj
+++ b/src/CSnakes.Runtime.Tests/CSnakes.Runtime.Tests.csproj
@@ -15,7 +15,7 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.10.0" />
-    <PackageReference Include="python" Version="3.12.4" />
+    <PackageReference Include="python" Version="$(PythonVersion)" />
     <PackageReference Include="xunit" Version="2.9.0" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.8.2">
       <PrivateAssets>all</PrivateAssets>

--- a/src/Directory.Build.props
+++ b/src/Directory.Build.props
@@ -1,37 +1,40 @@
 <Project>
-    <PropertyGroup>
-        <TargetFramework>net8.0</TargetFramework>
-        <Nullable>enable</Nullable>
-        <ImplicitUsings>enable</ImplicitUsings>
-        <LangVersion>latest</LangVersion>
-        <WarningsAsErrors>true</WarningsAsErrors>
-    </PropertyGroup>
-    <ItemGroup>
-        <None Include="..\..\docs\res\logo.jpeg">
-            <Pack>True</Pack>
-            <PackagePath>\</PackagePath>
-        </None>
-        <None Include="..\..\README.md">
-            <Pack>True</Pack>
-            <PackagePath>\</PackagePath>
-        </None>
-    </ItemGroup>
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)../'))" />
 
-    <PropertyGroup>
-        <Authors>Anthony Shaw, Aaron Powell</Authors>
-        <PackageProjectUrl>https://github.com/tonybaloney/CSnakes</PackageProjectUrl>
-        <PackageIcon>logo.jpeg</PackageIcon>
-        <PackageReadmeFile>README.md</PackageReadmeFile>
-        <RepositoryUrl>https://github.com/tonybaloney/CSnakes</RepositoryUrl>
-        <VersionPrefix>1.0.17</VersionPrefix>
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <LangVersion>latest</LangVersion>
+    <WarningsAsErrors>true</WarningsAsErrors>
+  </PropertyGroup>
 
-        <PublishRepositoryUrl>true</PublishRepositoryUrl>
-        <EmbedUntrackedSources>true</EmbedUntrackedSources>
-        <DebugType>embedded</DebugType>
-        <ContinuousIntegrationBuild Condition="'$(CI)' == 'true'">true</ContinuousIntegrationBuild>
-    </PropertyGroup>
+  <ItemGroup>
+    <None Include="..\..\docs\res\logo.jpeg">
+      <Pack>True</Pack>
+      <PackagePath>\</PackagePath>
+    </None>
+    <None Include="..\..\README.md">
+      <Pack>True</Pack>
+      <PackagePath>\</PackagePath>
+    </None>
+  </ItemGroup>
 
-    <PropertyGroup>
-        <EnableLocalPackaging>False</EnableLocalPackaging>
-    </PropertyGroup>
+  <PropertyGroup>
+    <Authors>Anthony Shaw, Aaron Powell</Authors>
+    <PackageProjectUrl>https://github.com/tonybaloney/CSnakes</PackageProjectUrl>
+    <PackageIcon>logo.jpeg</PackageIcon>
+    <PackageReadmeFile>README.md</PackageReadmeFile>
+    <RepositoryUrl>https://github.com/tonybaloney/CSnakes</RepositoryUrl>
+    <VersionPrefix>1.0.17</VersionPrefix>
+
+    <PublishRepositoryUrl>true</PublishRepositoryUrl>
+    <EmbedUntrackedSources>true</EmbedUntrackedSources>
+    <DebugType>embedded</DebugType>
+    <ContinuousIntegrationBuild Condition="'$(CI)' == 'true'">true</ContinuousIntegrationBuild>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <EnableLocalPackaging>False</EnableLocalPackaging>
+  </PropertyGroup>
 </Project>

--- a/src/Integration.Tests/Integration.Tests.csproj
+++ b/src/Integration.Tests/Integration.Tests.csproj
@@ -3,63 +3,8 @@
   <PropertyGroup>
     <IsPackable>false</IsPackable>
     <IsTestProject>true</IsTestProject>
-	<EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
-</PropertyGroup>
-
-  <ItemGroup>
-    <None Remove="python\requirements.txt" />
-    <None Remove="python\test_args.py" />
-    <None Remove="python\test_basic.py" />
-    <None Remove="python\test_defaults.py" />
-    <None Remove="python\test_dependency.py" />
-    <None Remove="python\test_dicts.py" />
-    <None Remove="python\test_exceptions.py" />
-    <None Remove="python\test_false_returns.py" />
-    <None Remove="python\test_generators.py" />
-    <None Remove="python\test_keywords.py" />
-    <None Remove="python\test_none.py" />
-    <None Remove="python\test_reserved.py" />
-    <None Remove="python\test_tuples.py" />
-  </ItemGroup>
-
-  <ItemGroup>
-    <AdditionalFiles Include="python\test_args.py">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
-    </AdditionalFiles>
-    <AdditionalFiles Include="python\test_basic.py">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
-    </AdditionalFiles>
-    <AdditionalFiles Include="python\test_defaults.py">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
-    </AdditionalFiles>
-    <AdditionalFiles Include="python\test_dependency.py">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
-    </AdditionalFiles>
-    <AdditionalFiles Include="python\test_dicts.py">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
-    </AdditionalFiles>
-    <AdditionalFiles Include="python\test_exceptions.py">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
-    </AdditionalFiles>
-    <AdditionalFiles Include="python\test_false_returns.py">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
-    </AdditionalFiles>
-    <AdditionalFiles Include="python\test_generators.py">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
-    </AdditionalFiles>
-    <AdditionalFiles Include="python\test_keywords.py">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
-    </AdditionalFiles>
-    <AdditionalFiles Include="python\test_none.py">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
-    </AdditionalFiles>
-    <AdditionalFiles Include="python\test_reserved.py">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
-    </AdditionalFiles>
-    <AdditionalFiles Include="python\test_tuples.py">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
-    </AdditionalFiles>
-  </ItemGroup>
+    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
+  </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="coverlet.collector" Version="6.0.0" />
@@ -68,7 +13,7 @@
     <PackageReference Include="System.Text.RegularExpressions" Version="4.3.1" />
     <PackageReference Include="xunit" Version="2.5.3" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.5.3" />
-    <PackageReference Include="python" Version="3.12.4" />
+    <PackageReference Include="python" Version="$(PythonVersion)" />
     <PackageReference Include="MartinCostello.Logging.XUnit" Version="0.4.0" />
   </ItemGroup>
 
@@ -77,15 +22,18 @@
     <Using Include="CSnakes.Runtime" />
   </ItemGroup>
 
-	<ItemGroup>
-		<ProjectReference Include="..\CSnakes.Runtime\CSnakes.Runtime.csproj" />
-		<ProjectReference Include="..\CSnakes\CSnakes.csproj" ReferenceOutputAssembly="false" OutputItemType="Analyzer" />
-	</ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\CSnakes.Runtime\CSnakes.Runtime.csproj" />
+    <ProjectReference Include="..\CSnakes\CSnakes.csproj" ReferenceOutputAssembly="false" OutputItemType="Analyzer" />
+  </ItemGroup>
 
-	<ItemGroup>
-		<Content Include="python\requirements.txt">
-		  <CopyToOutputDirectory>Always</CopyToOutputDirectory>
-		</Content>
-		<Content Include="xunit.runner.json" CopyToOutputDirectory="PreserveNewest" />
-	</ItemGroup>
+  <ItemGroup>
+    <AdditionalFiles Include="python\*.py">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </AdditionalFiles>
+    <Content Include="python\requirements.txt">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </Content>
+    <Content Include="xunit.runner.json" CopyToOutputDirectory="PreserveNewest" />
+  </ItemGroup>
 </Project>

--- a/src/Profile/Profile.csproj
+++ b/src/Profile/Profile.csproj
@@ -20,7 +20,7 @@
   <ItemGroup>
     <PackageReference Include="BenchmarkDotNet" Version="0.14.0" />
     <PackageReference Include="Microsoft.VisualStudio.DiagnosticsHub.BenchmarkDotNetDiagnosers" Version="17.12.35209.2" />
-    <PackageReference Include="python" Version="3.12.4" />
+    <PackageReference Include="python" Version="$(PythonVersion)" />
   </ItemGroup>
 
   <ItemGroup>

